scripts/src/java/org/oppia/android/scripts/common/MavenDependenciesListGenerator.kt

```diff
@@ -0,0 +1,247 @@
+   *
+   * @param pathToPbFile path to the pb file to be parsed
+   * @param proto instance of the proto class
+   * @return proto class from the parsed text proto file
+   */
+  private fun getProto(
+    pathToPbFile: String,
+    proto: MavenDependencyList
+  ): MavenDependencyList {
+    return FileInputStream(File(pathToPbFile)).use {
+      proto.newBuilderForType().mergeFrom(it)
+    }.build() as MavenDependencyList
+  }
+
+  fun genearateDependenciesListFromMavenInstall(
+    pathToMavenInstall: String,
+    bazelQueryDepsNames: List<String>
+  ): List<MavenListDependency> {
+    val dependencyTree = retrieveDependencyTree(pathToMavenInstall)
+    val mavenInstallDependencyList = dependencyTree.mavenListDependencies.dependencyList
+    return mavenInstallDependencyList.filter { dep ->
+      bazelQueryDepsNames.contains(omitVersionAndReplaceColonsHyphensPeriods(dep.coord))
+    }
+  }
+
+  private fun omitVersionAndReplaceColonsHyphensPeriods(artifactName: String): String {
+    return artifactName.substring(0, artifactName.lastIndexOf(':'))
+      .replace('.', '_')
+      .replace(':', '_')
+      .replace('-', '_')
+  }
+
+  fun retrieveDependencyListFromPom(
+    finalDependenciesList: List<MavenListDependency>
+  ): MavenDependencyList {
+    val mavenDependencyList = finalDependenciesList.map { it ->
+      // Remove ".jar" or ".aar" or any other extension from the specified url.
+      val pomFileUrl = "${it.url?.substringBeforeLast('.')}.pom"
+      val artifactName = it.coord
+      val artifactVersion = artifactName.substringAfterLast(':')
+      val pomFile = licenseFetcher.scrapeText(pomFileUrl)
+      val mavenDependency = MavenDependency.newBuilder().apply {
+        this.artifactName = it.coord
+        this.artifactVersion = artifactVersion.toString()
+        this.addAllLicense(extractLicenseLinksFromPom(pomFile))
+      }
+      mavenDependency.build()
+    }
+    return MavenDependencyList.newBuilder().addAllMavenDependency(mavenDependencyList).build()
+  }
+
+  private fun extractLicenseLinksFromPom(
+    pomText: String
+  ): List<License> {
+    val licenseList = mutableListOf<License>()
+    val builderFactory = DocumentBuilderFactory.newInstance()
+    val docBuilder = builderFactory.newDocumentBuilder()
+    val doc = docBuilder.parse(InputSource(pomText.byteInputStream()))
+
+    val licenses = doc.getElementsByTagName("license")
+    for (i in 0 until licenses.getLength()) {
+      if (licenses.item(0).getNodeType() == Node.ELEMENT_NODE) {
+        val element = licenses.item(i) as Element
+        val licenseName = getNodeValue("name", element)
+        val licenseLink = replaceHttpWithHttps(getNodeValue("url", element))
+        licenseList.add(
+          License.newBuilder().apply {
+            this.licenseName = licenseName
+            this.originalLink = licenseLink
+          }.build()
+        )
+      }
+    }
+    return licenseList
+  }
+
+  fun getDependencyListFromMavenInstall(
```

Comment:
move this function to line number 128 just to keep app public function together

Reply:
Done.